### PR TITLE
Attenuate fixed value loss and adjust weight in training jobs 

### PR DESCRIPTION
**Pull request by Rebecca-Qian**
# Description

This PR sets up gradient attenuation for fixed value loss:
- Separate Adam optimizer for fixed value head
- Remove fixed value loss from combined loss in backprop; apply gradient updates manually for shared decoder layers
- Configure fixed value loss attenuation factor from command line
- Save fixed value configs in model name generation (used to organize saved models)

Also changed some param types to store bool values for more convenient sweeps.

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

# Testing

Loss converges during some test training runs.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [x] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

**Review by kavyasrinet — APPROVED** [review]
You might want to rebase the previous PR on this one.
